Framework/Kernel/src/EnabledWhenProperty.cpp

```diff
@@ -37,10 +37,9 @@ EnabledWhenProperty::EnabledWhenProperty(
     : // This method allows the Python interface to easily construct these
       // objects
       // Copy the object then forward onto our move constructor
-      EnabledWhenProperty(
-          std::move(std::make_shared<EnabledWhenProperty>(conditionOne)),
-          std::move(std::make_shared<EnabledWhenProperty>(conditionTwo)),
-          logicOperator) {}
+      EnabledWhenProperty(std::make_shared<EnabledWhenProperty>(conditionOne),
+                          std::make_shared<EnabledWhenProperty>(conditionTwo),
```

Comment:
Historically we have used `boost::shared_ptr` because of its compatibility with `boost::python`.  Should these parameters (and related functions/classes) be changed to use `boost::shared_ptr`?